### PR TITLE
make disocvered policies work

### PR DESCRIPTION
**Pull request by achrefbensaad**
Signed-off-by: Achref ben saad <achref@accuknox.com>

### CODE REVIEW

File: src/systempolicy/systemPolicy.go [hunk 1]
```diff
@@ -788,6 +789,42 @@ func mergeSysPolicies(pols []types.KnoxSystemPolicy) []types.KnoxSystemPolicy {
...
+			if len(processDirRule.FromSource) == 0 {
+				gloablBinaries = append(gloablBinaries, processDirRule.Dir)
+			}
+		}
+	}
+
+	for _, pol := range results {
+		for i, filePath := range pol.Spec.File.MatchPaths {
+			for _, binary := range filePath.FromSource {
+				if slices.Contains(gloablBinaries, binary.Path) {
```
Comment (seswarrajan) [inline src/systempolicy/systemPolicy.go]: It should be something like strings.HasPrefix() for each path right. Are we not comparing to full fromSource path ?

File: src/systempolicy/systemPolicy.go [hunk 2]
```diff
@@ -788,6 +789,42 @@ func mergeSysPolicies(pols []types.KnoxSystemPolicy) []types.KnoxSystemPolicy {
 			})
 		}
 	}
+	// If a binary is allowed without fromSource, Convert the file access for it to global access
+
+	// Pre-compute: needed to optimize search
+
+	gloablBinaries := []string{}
```
Comment (Ankurk99) [inline src/systempolicy/systemPolicy.go]: globalBinaries*

File: src/systempolicy/systemPolicy.go (hunk 1 above)
Comment (achrefbensaad) [reply src/systempolicy/systemPolicy.go]: We dont need that because fromsource has absolute binary paths.

**Review by seswarrajan — APPROVED** [review]
LGTM

**Comment by achrefbensaad** [comment]
@seswarrajan , @Ankurk99  can you please re-review this.

**Pull request merged**